engine/src/test/java/org/camunda/bpm/engine/test/api/repository/DeleteProcessDefinitionTest.java

```diff
@@ -256,6 +258,31 @@ public void testDeleteProcessDefinitionAndRedeploy() {
     repositoryService.deleteDeployment(deployment2.getId(), true);
   }
 
+  @Test
+  public void shouldRestorePreviousStartTimerDefinitions() {
+    BpmnModelInstance processV1 = Bpmn.createExecutableProcess()
+        .id("one")
+        .startEvent()
+        .timerWithCycle("R/PT15M")
+        .userTask("aTaskName")
+        .endEvent()
+        .done();
+
+    testHelper.deploy(processV1);
+    testHelper.deploy("org/camunda/bpm/engine/test/repository/one.bpmn20.xml");
```

Comment:
🔧 While it's OK to mix those two possibilities (model API and resource), we usually try to stick with either of them in a test case if possible. You are free to choose which one you want to use, of course.